pkg/controllers/ippool/ippool_controller.go

```diff
@@ -47,24 +48,29 @@ type IPPoolReconciler struct {
 	Scheme     *apimachineryruntime.Scheme
 	Service    *ippool.IPPoolService
 	VPCService servicecommon.VPCServiceProvider
+	Recorder   record.EventRecorder
 }
 
-func deleteSuccess(r *IPPoolReconciler, _ *context.Context, _ *v1alpha2.IPPool) {
+func deleteSuccess(r *IPPoolReconciler, _ *context.Context, o *v1alpha2.IPPool) {
+	r.Recorder.Event(o, v1.EventTypeNormal, "SuccessfulDelete", "SuccessfulDelete: IPPool CR has been successfully deleted")
 	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteSuccessTotal, MetricResType)
 }
 
 func deleteFail(r *IPPoolReconciler, c *context.Context, o *v1alpha2.IPPool, e *error) {
 	r.setReadyStatusFalse(c, o, metav1.Now(), e)
+	r.Recorder.Event(o, v1.EventTypeWarning, "FailDelete", fmt.Sprintf("FailError: %v", *e))
 	metrics.CounterInc(r.Service.NSXConfig, metrics.ControllerDeleteFailTotal, MetricResType)
```

Comment:
could you put reason in const?  in Message, just "Error:" since the reason is Faildelete